### PR TITLE
Add compression to protocol 

### PR DESCRIPTION
**Pull request by justus-camp**
## Description
This is a continuation of #11549 where I'm adding a compression property to the protocol. I've moved the changes to main to propagate them to ordering services faster as requested in the referenced PR.

### CODE REVIEW

File: server/routerlicious/packages/lambdas/src/deli/lambda.ts [hunk 1]
```diff
@@ -1194,7 +1194,8 @@ export class DeliLambda extends TypedEventEmitter<IDeliLambdaEvents> implements
             timestamp: message.timestamp,
             traces: message.operation.traces,
             type: message.operation.type,
-        };
+            compression: (message.operation as any).compression,
```
Comment (vladsud) [inline server/routerlicious/packages/lambdas/src/deli/lambda.ts:1197]: Please add "// back-compat" comment here. It's not the best (i.e. it's less about compat), but we do scan periodically repo for these comments and remove code that is no longer needed (including unneeded casts). Or open an issue to track removal of it.

File: server/routerlicious/packages/test-utils/src/messageFactory.ts [hunk 2]
```diff
@@ -84,7 +84,8 @@ export class MessageFactory {
             referenceSequenceNumber,
             traces: [],
             type,
-        };
+            compression: undefined,
```
Comment (vladsud) [inline server/routerlicious/packages/test-utils/src/messageFactory.ts:87]: This should not be needed, right?
It probably does not matter that much, but I observed similar change to be costly in tight loop:

https://github.com/microsoft/FluidFramework/pull/12058/files#diff-4955d937755632e07ceb6cff6c215904ae4e7cc8ceacdbf801868e3188760594R372

**Review by vladsud — APPROVED** [review]

File: server/routerlicious/packages/test-utils/src/messageFactory.ts (hunk 2 above)
Comment (justus-camp) [reply server/routerlicious/packages/test-utils/src/messageFactory.ts]: We get two failing tests if I don't include this change where the tests are checking that the socket receives the same message that the document factory generates. I could change the tests instead (there are only two) or I can keep this change and the tests will pass. Which do we prefer?

Comment (vladsud) [reply server/routerlicious/packages/test-utils/src/messageFactory.ts]: Or, that's test code - I guess it does not matter :) Putting undefined is fine here

**Pull request merged**